File: CaseStudies/glass/Implementations/C/src/InputConstraints.c
#include "InputConstraints.h"

#include "InputParameters.h"

#include <stdlib.h>
#include <stdio.h>

void check_constraints(InputParameters* params) {
    if (params->a <= 0 || params->b <= 0) {
        printf("InputError: a and b must be greater than 0");
        exit(EXIT_FAILURE);
    }
    if (params->asprat < 1 || params->asprat > 5) {
        printf("InputError: a/b must be between 1 and 5");
        exit(EXIT_FAILURE);
    }
    if (!(  (params->t == 2.50) 
         || (params->t == 2.70) 
         || (params->t == 3.00) 
         || (params->t == 4.00) 
         || (params->t == 5.00) 
         || (params->t == 6.00)
         || (params->t == 8.00) 
         || (params->t == 10.00) 
         || (params->t == 12.00) 
         || (params->t == 16.00) 
         || (params->t == 19.00) 
         || (params->t == 22.00) ) ) {
        printf("InputError: t must be in [2.5,2.7,3.0,4.0,5.0,6.0,8.0,10.0,12.0,16.0,19.0,22.0]");
        exit(EXIT_FAILURE);
    }
    if (params->tnt <= 0) {
        printf("InputError: TNT must be greater than 0");
        exit(EXIT_FAILURE);
    }
    if (params->wtnt < 4.5 || params->wtnt > 910) {
        printf("InputError: wtnt must be between 4.5 and 910");
        exit(EXIT_FAILURE);
    }
    if (params->sd < 6 || params-> sd > 130) {
        printf("InputError: SD must be between 6 and 130");
        exit(EXIT_FAILURE);
    }
}

File: Presentations/WG2_11/GlassBR_Code/fromGOOL/python/InputConstraints.py
from __future__ import print_function
import sys
import math
import InputParameters


def check_constraints(inparams):
    if (inparams.a <= 0.0) :
        raise Exception("InputError: a must be greater than 0")
    if (inparams.b <= 0.0) :
        raise Exception("InputError: b must be greater than 0")
    if (inparams.asprat < 1.0) :
        raise Exception("InputError: a/b cannot be less than 1.0")
    if (inparams.asprat > 5.0) :
        raise Exception("InputError: a/b cannot be greater than 5.0")
    if (not((((((((((((inparams.t == 2.5) or (inparams.t == 2.7)) or (inparams.t == 3.0)) or (inparams.t == 4.0)) or (inparams.t == 5.0)) or (inparams.t == 6.0)) or (inparams.t == 8.0)) or (inparams.t == 10.0)) or (inparams.t == 12.0)) or (inparams.t == 16.0)) or (inparams.t == 19.0)) or (inparams.t == 22.0))) :
        raise Exception("InputError: t must be in [2.5,2.7,3.0,4.0,5.0,6.0,8.0,10.0,12.0,16.0,19.0,22.0]")
    if (inparams.tnt <= 0.0) :
        raise Exception("InputError: tnt must be greater than 0")
    if (inparams.wtnt < 4.5) :
        raise Exception("InputError: wtnt cannot be less than 4.5")
    if (inparams.wtnt > 910.0) :
        raise Exception("InputError: wtnt cannot be greater than 910.0")
    if (inparams.sd < 6.0) :
        raise Exception("InputError: sd cannot be less than 6.0")
    if (inparams.sd > 130.0) :
        raise Exception("InputError: sd cannot be greater than 130.0")


